index.js

```diff
@@ -53,6 +53,23 @@ function start(func, port, cb, options) {
       payload.on('error', done);
     });
 
+  server.addContentTypeParser('*',
+    function(_, payload, done) {
+      let data;
+      payload.on('data', chunk => { data += chunk; });
+      payload.on('end', () => {
+        done(null, data);
+      });
+    }
+  );
+
+  server.addHook('preValidation', (request, reply, done) => {
+    if (request.headers['content-type'] === '') {
```

Comment:
Wouldn't it be better to just remove the header in this case instead of setting it to an (arbitrary) content type, that might cause issues later on (because the content type probably does not fit the real content type) ? I guess if an empty content-type is specified it just should be treated as _no content type given_

Reply:
Removing the header doesn't seems to work. Defaulting it to `application/octet-stream` seems to work better.

Reply:
https://github.com/cloudevents/spec/issues/614 hmm, it shouldn't default to `application/octet-stream` but `application/json`

Reply:
OK it appears that `CloudEvent SDK` defaults to JSON when `content-type` absent. I am not sure if that's good idea, IMHO `application/octet-stream` would be better. We can workaround this by setting header here before it's passed to `CloudEvents SDK`.

Reply:
@lance ^^^ wdyt?